protocols/request-response/src/lib.rs

```diff
@@ -169,6 +169,8 @@ pub enum Event<TRequest, TResponse, TChannelResponse = TResponse> {
 pub enum OutboundFailure {
     /// The request could not be sent because a dialing attempt failed.
     DialFailure,
+    /// The request could not be sent because maximum concurrent streams reached.
+    MaxStreamsReached,
```

Comment:
Ideally, we solve this with backpressure from the behaviour to the handler. I'd rather not have us introduce a variant here that will be removed against once somebody implements backpressure for `request-response`. Can we just report an IO error instead?

That would also make this a non-breaking change.

Reply:
I decided to implement an alternative approach after figuring out that the following can also cause code to hang:

https://github.com/libp2p/rust-libp2p/blob/94fef37d9b2e66c25a2baab87ce74ac1adeee4fd/protocols/request-response/src/handler.rs#L238-L244

I will open another PR that fixes both issues without a breaking change.

Reply:
https://github.com/libp2p/rust-libp2p/pull/5419

Reply:
thanks for driving this @oblique I agree with Thomas on https://github.com/libp2p/rust-libp2p/pull/5419#issuecomment-2132435880 can we just report an IO error?

Reply:
Changed it to IO error